lib/src/analyzers/unused_code_analyzer/public_code_visitor.dart

```diff
@@ -0,0 +1,53 @@
+// ignore_for_file: public_member_api_docs
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+
+class PublicCodeVisitor extends GeneralizingAstVisitor<void> {
+  final Set<Element> topLevelElements = {};
+
+  PublicCodeVisitor();
+
+  @override
+  void visitCompilationUnitMember(CompilationUnitMember node) {
+    if (node is FunctionDeclaration) {
+      if (node.name.name == 'main' || _hasPragmaAnnotation(node.metadata)) {
```

Comment:
@incendial do we use similar code for detect unused files?

Reply:
Moved to utils